### DISCUSSION TITLE
Parcel2 - Yarn workspaces

### DISCUSSION THREAD

**Discussion by klemenoslaj** [parcel-bundler/parcel#4651] [General] [answered]
I just wanted to ask if parcel2 intends to support yarn workspaces out of the box?
I am playing around with it out right now. While it mostly works fine I did find several issues, but they're all hard to pin down.

Issues:
- sass `includePaths` not working then `source` is in package.json (works fine without as then it looks into compiled bundle
- declaration files are not generated for workspace packages (not sure if yarn related or issue overall)
- often times library bundle (parcel build index.ts - not html) is corrupt, unless `--no-scope-hoist` is enabled

The setup includes two transformers that are not the default:
- "@parcel/transformer-typescript-tsc": "^2.0.0-nightly.277",
- "@parcel/transformer-sass": "^2.0.0-nightly.277", (only version is not default)

Before I go on making issues I just wanted to make sure this is something you guys want to support. Given monorepos are gaining traction I do believe so, but still.

**Comment by mischnic** [comment, marked as answer]
> I just wanted to ask if parcel2 intends to support yarn workspaces out of the box?

Absolutely

**Reply by klemenoslaj** [reply]
Btw, did I assume right? If `source` is in package.json, parcel will use the sources, not the `main`? I am observing such a behavior, that's why I ask. However, when `main` is not there, it seems to choke.

**Reply by mischnic** [reply]
Not sure what you mean.

```json
{
	"source": "src/index.js",
	"main": "dist/index.js"
}
```
Then `parcel build .` will use "source" and be equivalent to `parcel build src/index.js`.

So main is the output and source is the input.

**Reply by klemenoslaj** [reply]
Yes, I mean I terms of workspaces.

Package A
```json
{
  "source": "./src/index.ts",
  "main": "./dist/index.js"
}
```

Now I have an application that consumes `Package A` trough yarn workspaces linking mechanism, I had a feeling that parcel was taking `source` all the time to compile application bundle and not the main.

---

If that is not the case, I wonder how the following happens - when I have a library containing some sass dependency, the application bundle breaks as it cannot find that dependency - pointing with an error to my actual scss source file, not into dist. If it would simply take the dist it's already all there, right? Now I wonder, why would it even go looking into actual sources?

**Reply by mischnic** [reply]
> I had a feeling that parcel was taking source all the time to compile application bundle and not the main.

Yes, that is correct. 

The idea is that when working in a monorepo, you're probably:
- working on multiple packages at a time, so rebuilding them on every change would be cumbersome
- not building/transpiling the packages at all, so just using `main` would probably result in something like  `Cannot resolve "dist/index.js"`.

(And please do create issues with exampleS for your original three problems)

**Reply by klemenoslaj** [reply]
Ok super cool. Was about to suggest exactly this, if it's not like that yet.. 

> working on multiple packages at a time, so rebuilding them on every change would be cumbersome

I do wonder if Parcel should take `main` from dist, if it exists? 🤔
Theoretically, bundling the application with library from sources or pre-built bundle might poses some risks as it's not really really the same :) Just a though.

_I'll create a monorepo with the issues later today, hopefully will get the time. Thanks for you super quick responses_
